src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -503,8 +520,60 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 TimeSpan timerInterval = timer == null ? TimeSpan.MinValue : TimeSpan.FromMilliseconds(timer.Interval);
                 try
                 {
-                    await InvokeAsync(invoker, parameterHelper, timeoutTokenSource, functionCancellationTokenSource,
-                    throwOnTimeout, timerInterval, instance);
+                    // A variable for the function result
+                    FunctionResult functionResult = null;
+
+                    // Create a dictionary that will pass the user properties from the Executing filter to the Executed filter
+                    Dictionary<string, object> properties = new Dictionary<string, object>();
+
+                    // Create the context objects for the filter execution
+                    FunctionExecutingContext executingContext = new FunctionExecutingContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger);
+                    executingContext.JobHost = jobHost;
+                    executingContext.Config = config;
+
+                    FunctionExecutedContext executedContext = new FunctionExecutedContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger, functionResult);
+                    executedContext.JobHost = jobHost;
+                    executedContext.Config = config;
+
+                    // Lambda function for function invocation
+                    Func<Task> invokeAsync = async () =>
+                    {
+                        await InvokeAsync(invoker, parameterHelper, timeoutTokenSource, functionCancellationTokenSource,
+                            throwOnTimeout, timerInterval, instance);
+                    };
+
+                    // Lambda function that wraps filters around the function
+                    Func<Func<Task>, IFunctionInvocationFilter, Func<Task>> wrapFunctionWithFilter = (innerFunction, filter) =>
+                    {
+                        Func<Task> filteredFunction = async () =>
+                        {
+                            try
+                            {
+                                await filter.OnExecutingAsync(executingContext, functionCancellationTokenSource.Token);
+                                await innerFunction();
+                                await filter.OnExecutedAsync(executedContext, functionCancellationTokenSource.Token);
+                            }
+                            catch (Exception e)
+                            {
+                                // How should the exception results be cached?
+                                executedContext.Result = new FunctionResult(e);
```

Comment:
Update: aspnet doesn't wrap multiple exceptions in AggregateException, thanks @fabiocav for testing that.

In that case, we want to go with my below comment of capturing the last exception in an ExceptionDispatchInfo on the filter ExecutedContext